common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java

```diff
@@ -0,0 +1,292 @@
+import muramasa.antimatter.capability.machine.MachineRecipeHandler;
+import muramasa.antimatter.machine.MachineState;
+import muramasa.antimatter.machine.event.ContentEvent;
+import muramasa.antimatter.machine.types.Machine;
+import muramasa.antimatter.tile.TileEntityMachine;
+import muramasa.antimatter.util.Utils;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.inventory.ContainerData;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Explosion;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static muramasa.antimatter.machine.MachineState.ACTIVE;
+import static muramasa.antimatter.machine.MachineState.IDLE;
+import static muramasa.antimatter.machine.Tier.BRONZE;
+import static muramasa.gregtech.data.Materials.DistilledWater;
+import static muramasa.gregtech.data.Materials.Steam;
+import static net.minecraftforge.fluids.capability.IFluidHandler.FluidAction.EXECUTE;
+
+public class TileEntityLavaBoiler extends TileEntityMachine<TileEntityLavaBoiler> {
+    int maxHeat = 500, heat, fuel = 0, maxFuel, lossTimer = 0;
+    boolean hadNoWater;
+
+    public TileEntityLavaBoiler(Machine<?> type, BlockPos pos, BlockState state) {
+        super(type, pos, state);
+        fluidHandler.set(() -> new LavaBoilerFluidHandler(this));
+        recipeHandler.set(() -> new LavaBoilerRecipeHandler(this));
+    }
+
+    public int getFuel() {
+        return recipeHandler.map(r -> (((LavaBoilerRecipeHandler) r).getFuel())).orElse(0);
+    }
+
+    public int getHeat() {
+        return recipeHandler.map(r -> (((LavaBoilerRecipeHandler) r).getHeat())).orElse(0);
+    }
+
+    public int getMaxFuel() {
+        return recipeHandler.map(r -> (((LavaBoilerRecipeHandler) r).getMaxFuel())).orElse(0);
+
+    }
+
+    public int getMaxHeat() {
+        return recipeHandler.map(r -> (((LavaBoilerRecipeHandler) r).getMaxHeat())).orElse(0);
+    }
+
+    public static class LavaBoilerRecipeHandler extends MachineRecipeHandler<TileEntityLavaBoiler> {
+        int maxHeat = 500, heat, fuel = 0, maxFuel, lossTimer = 0;
+        boolean hadNoWater;
+
+        protected final ContainerData GUI_SYNC_DATA2 = new ContainerData() {
```

Comment:
do you need this, if you can use the built in gui syncing system?

Reply:
As above I used the existing coal boiler as a reference so I'm happy to change as needed, just need a bit of direction or something to use as a reference as I've not done much GUI stuff yet.

Reply:
Removed the GUI_SYNC_DATA2 block